src/utils/hashes/index.ts

```diff
@@ -62,13 +65,37 @@ const addLengthPrefix = (hex: string): string => {
   throw new Error("Variable integer overflow.");
 };
 
-export const computeBinaryTransactionHash = (txBlobHex: string): string => {
-  const prefix = HashPrefix.TRANSACTION_ID.toString(16).toUpperCase();
-  return sha512Half(prefix + txBlobHex);
-};
+/**
+ * Hashes the Transaction object as the ledger does. Throws if the transaction is unsigned.
+ *
+ * @param {Transaction | string} tx - A transaction to hash. Tx may be in binary blob form. Tx must be signed.
+ * @returns {string} A hash of tx.
+ * @throws {ValidationError} If the Transaction is unsigned.
```

Comment:
This is the incorrect JSdocs format. You can run the linter on just the files you're messing with using `eslint path/to/file`